Unrecognized field "max_consecutive_failed_dag_runs" in DAG model

### Apache Airflow version

2.9.1

### If "Other Airflow 2 version" selected, which one?

_No response_

### What happened?

After upgrading Airflow from `2.8.4` to `2.9.1` we started to see serialization errors in our java client because of broken Rest API contract:
```
Caused by: com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "max_consecutive_failed_dag_runs" (class com.dremio.daas.workflowservice.airflow.client.model.DAG), not marked as ignorable (27 known properties: "fileloc", "max_active_runs", "next_dagrun_create_after", "next_dagrun", "max_active_tasks", "last_parsed_time", "tags", "timetable_description", "pickle_id", "last_pickled", "is_paused", "next_dagrun_data_interval_end", "next_dagrun_data_interval_start", "root_dag_id", "scheduler_lock", "dag_id", "file_token", "is_active", "last_expired", "dag_display_name", "is_subdag", "has_task_concurrency_limits", "owners", "default_view", "schedule_interval", "description", "has_import_errors"])
 at [Source: (org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$UnCloseableInputStream); line: 42, column: 2] (through reference chain: com.dremio.daas.workflowservice.airflow.client.model.DAG["max_consecutive_failed_dag_runs"])
	at com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException.from(UnrecognizedPropertyException.java:61)
	at com.fasterxml.jackson.databind.DeserializationContext.handleUnknownProperty(DeserializationContext.java:1138)
	at com.fasterxml.jackson.databind.deser.std.StdDeserializer.handleUnknownProperty(StdDeserializer.java:2224)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownProperty(BeanDeserializerBase.java:1709)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownProperties(BeanDeserializerBase.java:1659)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:460)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1409)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:352)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:185)
	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:323)
	at com.fasterxml.jackson.databind.ObjectReader._bind(ObjectReader.java:2079)
	at com.fasterxml.jackson.databind.ObjectReader.readValue(ObjectReader.java:1229)
	at org.glassfish.jersey.jackson.internal.jackson.jaxrs.base.ProviderBase.readFrom(ProviderBase.java:829)
	at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$TerminalReaderInterceptor.invokeReadFrom(ReaderInterceptorExecutor.java:233)
	at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$TerminalReaderInterceptor.aroundReadFrom(ReaderInterceptorExecutor.java:212)
	at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor.proceed(ReaderInterceptorExecutor.java:132)
	at org.glassfish.jersey.message.internal.MessageBodyFactory.readFrom(MessageBodyFactory.java:1072)
	at org.glassfish.jersey.message.internal.InboundMessageContext.readEntity(InboundMessageContext.java:919)
	... 42 more
```
We generate client stub from the following Open API [contract](https://raw.githubusercontent.com/apache/airflow/2.9.1/airflow/api_connexion/openapi/v1.yaml).

### What you think should happen instead?

Seems like starting from `2.9.0` the new [max_consecutive_failed_dag_runs](https://github.com/apache/airflow/blob/7163da9a714e9dca6e2b2322fcb928562ce2e237/airflow/models/dag.py#L3650) field has been added into DAG model.
However this field is missing in [Open API contract](https://raw.githubusercontent.com/apache/airflow/2.9.1/airflow/api_connexion/openapi/v1.yaml).
We need to add it there. Probably there is a missing test coverage for such API change.
P.S. Maybe it is somehow related to [this](https://github.com/apache/airflow/blob/7163da9a714e9dca6e2b2322fcb928562ce2e237/airflow/config_templates/config.yml#L119) field.

### How to reproduce

When we use the `openapi-generator-maven-plugin`, it generates the java client from the [openapi yaml file](https://raw.githubusercontent.com/apache/airflow/2.9.1/airflow/api_connexion/openapi/v1.yaml) with missing `max_consecutive_failed_dag_runs` field.
This field is present in REST API response:
```
curl --location 'http://localhost:8080/api/v1/dags/my_dag' \ 
--header 'Authorization: Basic YWRtaW46aEhEMzduUTdjU2Nodk5QSA==' | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1221  100  1221    0     0  15026      0 --:--:-- --:--:-- --:--:-- 15074
{
  "dag_display_name": "my_dag",
  "dag_id": "my_dag",
  "default_view": "grid",
  "description": null,
  "file_token": ".eJwdyssNgEAIBcBeLMBXjWdDBJW4yAbWX_car5PpMKRE4izEahq4PDZMzgIOMXUwUf46F7--e-okqE9bfYfcZLVIwvzYG9SqRxMemZbs69O9pd8kRA.B3iHbUJ8HsCW0qD1uY7-vD",
  "fileloc": "/my_path/imported_dags.py",
  "has_import_errors": false,
  "has_task_concurrency_limits": false,
  "is_active": true,
  "is_paused": false,
  "is_subdag": false,
  "last_expired": null,
  "last_parsed_time": "2024-05-24T15:10:36.767525+00:00",
  "last_pickled": null,
  "max_active_runs": 100,
  "max_active_tasks": 16,
  "max_consecutive_failed_dag_runs": 0,
  "next_dagrun": null,
  "next_dagrun_create_after": null,
  "next_dagrun_data_interval_end": null,
  "next_dagrun_data_interval_start": null,
  "owners": [
    "airflow"
  ],
  "pickle_id": null,
  "root_dag_id": null,
  "schedule_interval": null,
  "scheduler_lock": null,
  "tags": [
    {
      "name": "static"
    }
  ],
  "timetable_description": "Never, external triggers only"
}
``` 
Generated Java Client has a following fields:
```
@JsonCreator
  public DAG(
    @JsonProperty(JSON_PROPERTY_DAG_ID) String dagId, 
    @JsonProperty(JSON_PROPERTY_DAG_DISPLAY_NAME) String dagDisplayName, 
    @JsonProperty(JSON_PROPERTY_ROOT_DAG_ID) String rootDagId, 
    @JsonProperty(JSON_PROPERTY_IS_ACTIVE) Boolean isActive, 
    @JsonProperty(JSON_PROPERTY_IS_SUBDAG) Boolean isSubdag, 
    @JsonProperty(JSON_PROPERTY_LAST_PARSED_TIME) OffsetDateTime lastParsedTime, 
    @JsonProperty(JSON_PROPERTY_LAST_PICKLED) OffsetDateTime lastPickled, 
    @JsonProperty(JSON_PROPERTY_LAST_EXPIRED) OffsetDateTime lastExpired, 
    @JsonProperty(JSON_PROPERTY_SCHEDULER_LOCK) Boolean schedulerLock, 
    @JsonProperty(JSON_PROPERTY_PICKLE_ID) String pickleId, 
    @JsonProperty(JSON_PROPERTY_DEFAULT_VIEW) String defaultView, 
    @JsonProperty(JSON_PROPERTY_FILELOC) String fileloc, 
    @JsonProperty(JSON_PROPERTY_FILE_TOKEN) String fileToken, 
    @JsonProperty(JSON_PROPERTY_OWNERS) List<String> owners, 
    @JsonProperty(JSON_PROPERTY_DESCRIPTION) String description, 
    @JsonProperty(JSON_PROPERTY_TIMETABLE_DESCRIPTION) String timetableDescription, 
    @JsonProperty(JSON_PROPERTY_TAGS) List<Tag> tags, 
    @JsonProperty(JSON_PROPERTY_MAX_ACTIVE_TASKS) Integer maxActiveTasks, 
    @JsonProperty(JSON_PROPERTY_MAX_ACTIVE_RUNS) Integer maxActiveRuns, 
    @JsonProperty(JSON_PROPERTY_HAS_TASK_CONCURRENCY_LIMITS) Boolean hasTaskConcurrencyLimits, 
    @JsonProperty(JSON_PROPERTY_HAS_IMPORT_ERRORS) Boolean hasImportErrors, 
    @JsonProperty(JSON_PROPERTY_NEXT_DAGRUN) OffsetDateTime nextDagrun, 
    @JsonProperty(JSON_PROPERTY_NEXT_DAGRUN_DATA_INTERVAL_START) OffsetDateTime nextDagrunDataIntervalStart, 
    @JsonProperty(JSON_PROPERTY_NEXT_DAGRUN_DATA_INTERVAL_END) OffsetDateTime nextDagrunDataIntervalEnd, 
    @JsonProperty(JSON_PROPERTY_NEXT_DAGRUN_CREATE_AFTER) OffsetDateTime nextDagrunCreateAfter
  ) {
```

### Operating System

Debian GNU/Linux 12 (bookworm)

### Versions of Apache Airflow Providers

apache-airflow-providers-celery==3.6.2
apache-airflow-providers-common-io==1.3.1
apache-airflow-providers-common-sql==1.12.0
apache-airflow-providers-datadog==3.5.1
apache-airflow-providers-fab==1.0.4
apache-airflow-providers-ftp==3.8.0
apache-airflow-providers-google==10.17.0
apache-airflow-providers-http==4.10.1
apache-airflow-providers-imap==3.5.0
apache-airflow-providers-postgres==5.10.2
apache-airflow-providers-slack==8.6.2
apache-airflow-providers-smtp==1.6.1
apache-airflow-providers-sqlite==3.7.1

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

_No response_

### Anything else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

--- a/airflow/api_connexion/openapi/v1.yaml
+++ b/airflow/api_connexion/openapi/v1.yaml
@@ -3123,6 +3123,14 @@ components:
             Earliest time at which this ``next_dagrun`` can be created.
 
             *New in version 2.3.0*
+        max_consecutive_failed_dag_runs:
+          type: integer
+          nullable: true
+          readOnly: true
+          description: |
+            (experimental) The maximum number of consecutive DAG failures before DAG is automatically paused.
+
+            *New in version 2.9.0*
 
     DAGCollection:
       description: |

--- a/airflow/www/static/js/types/api-generated.ts
+++ b/airflow/www/static/js/types/api-generated.ts
@@ -1116,6 +1116,12 @@ export interface components {
        * *New in version 2.3.0*
        */
       next_dagrun_create_after?: string | null;
+      /**
+       * @description (experimental) The maximum number of consecutive DAG failures before DAG is automatically paused.
+       *
+       * *New in version 2.9.0*
+       */
+      max_consecutive_failed_dag_runs?: number | null;
     };
     /**
      * @description Collection of DAGs.

